tools/build_tar/buildtar.go

```diff
@@ -559,6 +571,38 @@ func (f *fileMeta) getMode(fname string) os.FileMode {
 	return f.defaultMode
 }
 
+type timeString time.Time
+
+var y2k = time.Date(2000, 1, 1, 0, 0, 0, 0, time.UTC)
+
+const portableTime = "portable"
+
+func (ts *timeString) String() string {
+	t := time.Time(*ts)
+	if t.Equal(y2k) {
+		return portableTime
+	}
+	return strconv.FormatInt(t.Unix(), 10)
+}
+
+func (ts *timeString) Set(v string) error {
+	var t time.Time
+	switch v {
+	case "":
+		// Use zero
```

Comment:
also the zero value of time.Time is not the same as unix epoch 0, apparently?

```golang
	var t time.Time
	fmt.Printf("%s\n", t)
```
yields `0001-01-01 00:00:00 +0000 UTC`

Reply:
```golang
	var t time.Time
	fmt.Printf("%d\n", t.Unix())
```
-> `-62135596800`

who knew?

(I think unix epoch is probably a more expected "0" value?)

Reply:
Does not matter

Reply:
it does if we're wanting to maintain existing behavior.

Reply:
It doesn't, and that would change the behavior

Reply:
And the point is to get things done so you can move onto the next thing, not spend forever asymptotically approaching a local maximum.

Reply:
the current implementation of `build_tar` doesn't set the modification time, so tarballs implicitly use unix epoch.

this new implementation would've changed the default for older bazel (such as 0.23, which don't pass the `--mtime` flag) to use 0001-01-01, which is wildly different.

Reply:
That's now what setting `Header.ModTime = time.Time{}` does

Reply:
ah, interesting. clearly I did not know that.